### ISSUE TITLE
Toolbox Orginization

### ISSUE THREAD

**Issue by dube1503** [txst-per-group/Glowscript-Blocks#125]
_As a basic user, I should be able to determine the contents of each category with a quick glance._

**Comment by dube1503** [comment]
### Math Category Organization ###
![img_0308](https://user-images.githubusercontent.com/19582167/34846437-d64b50e8-f6dd-11e7-8b80-edd0f96c104b.JPG)
![img_0309](https://user-images.githubusercontent.com/19582167/34846446-d90561b6-f6dd-11e7-9b95-aba496bf7ff7.JPG)

Two options here for reorganizing the Math category. Both options include merging Vector into the math category but to avoid cluttering breaking them into sub categories. The first, all blocks existing within one flyout with labels and the second being a category tree (only of the options viably tested) where each category has the same respective blocks but exist under the "Math" label as a whole.

@eah13 @codestar12 @hgclose

**Comment by eah13** [comment]
Nice work. 

I'd lean towards the first option primarily because it's more browseable. The labels help.me group the blocks but I can scroll thru the whole thing, learning about what math blocks there are as I go. 

With the second option, i need to know or guess what the block I'm looking for is categorized. This can be more efficient once you know what you're looking for.  But I think less conducive to newer users finding what they want since new users will be primarily browsing instead of navigating to a known item. 

I hope I didnt over analyze here, but that's my thought process on why option 1 seems like a better fit for our user population.

**Comment by dube1503** [comment]
Definitely not over analyzing, would prefer a good consideration on the options before moving forward. 

I was leaning more towards the category tree because it was less dense and less crowded information wise, but you do have a very good point that it would create a time waste for newer users who are not sure which category they need to be looking in whereas the first option gives a much quicker dive into all available options.

Some AB testing on a small number may be helpful, but I think those are good enough points to look into how to actually implement the labeling feature (as of our current fork from blockly, we do not have the labeling toolbox feature.)

**Comment by codestar12** [comment]
@eah13 I agree the first one looks nicer; However, we estimate it would take us around 80 man hours to get labels that look like that (which would come with additional side benefits like buttons and dynamically created variables). The 2nd option would take about an hour. The question then is, "is it worth the extra effort".

**Comment by eah13** [comment]
Wow, that's a big difference!

I'm not certain that the second design is a net gain. Subdividing the math blocks so granularly seems like it could lead to confusion in the ways I discussed above. 

Are there design goals the second design would solve that seem worth the cognitive overhead for users who don't know which subcategory their block is in? If not, I'd vote for waiting for the expected effort of the first solution to decrease (via eureka moments or upstream insights) and/or the expected utility to users to increase.
